### PR TITLE
MDX image loader fix for absolute URLs

### PR DESCRIPTION
**Pull request by qburst-aparna**

### CODE REVIEW

File: packages/website/custom-plugins/mdx-image-loader.js [hunk 1]
```diff
@@ -9,43 +9,44 @@ module.exports = () => tree =>
   visit(tree, 'root', rNode => {
     visit(tree, 'image', node => {
       const text = node.url ? node.url.trim() : '';
+      if (text.startsWith('./') || text.startsWith('../')) {
```
Comment (Blasz) [inline packages/website/custom-plugins/mdx-image-loader.js]: I've noticed we're using `startsWith` in a couple of places, do we ship any polyfills for browsers that don't support this (IE11)?

**Comment by Blasz** [comment]
What's the context behind excluding absolute URLs in our mdx image loader?

File: packages/website/custom-plugins/mdx-image-loader.js (hunk 1 above)
Comment (Noviny) [reply packages/website/custom-plugins/mdx-image-loader.js]: We don't - I think we don't target ie11.

But also, this is node-only code so here it won't cause us any grief.

**Comment by Noviny** [comment]
> What's the context behind excluding absolute URLs in our mdx image loader?

I guess context of why this transformer exists, and then why it shouldn't be for everything.

By default, `img markdown` is transformed into `` by remark. This implies that images are available on the website at a relative location.

Next however doesn't do anything with pulling in images from relative paths, so all relative images are just absent/broken, as they will look in urls that don't exist.

If you transform it into mdx of

```mdx
import imgSrc from './somewhere.png'

<img src={imgSrc} />
```

next takes the mdx, and its image loader plugin will find and add the image to the website correctly. Neat! Win

The bug we had was we were transforming all images, so `text` was also being transformed into:

```mdx
import imgSrc from 'https://go/tangerine/favicon.png'

<img src={imgSrc} />
```

which doesn't work.

This PR checks whether it's something that we need to transform, and only modifies it when we need to.

**Pull request merged**